docs/software_layer/contribution_policy.md

```diff
@@ -0,0 +1,80 @@
+# Contribution policy
+
+When [openining a pull request to add software to EESSI](adding_software.md), the following requirements must
+be taken into account.
+
+!!! note
+
+    These requirements are subject to change, please check back regularly.
+
+## Open source software
+
+Only **open source software** can be added to the EESSI repository.
+
+Make sure that you are aware of software license, and that redistribution is allowed.
+
+For more information about a specific license, see the [SPDX license list](https://spdx.org/licenses/).
+
+!!! note
+
+    We intend to automatically verify that this requirement is met,
+    by requiring that the [SPDX license identifier](https://spdx.dev/ids/) is provided for all software.
+
+## Built by the bot
+
+All software included in the EESSI repository *must* be built autonomously by [our bot :robot:](../bot.md),
+see also the [semi-automatic software installation procedure](adding_software.md).
+
+## Supported by EasyBuild
+
+Currently, we require that all software being added to EESSI is supported by the *EasyBuild release* being used
+to perform the installation.
+
+That is, the easyconfig files used for the installation *must be included in the EasyBuild release*.
+
+We do allow the use of [`--from-pr`](https://docs.easybuild.io/integration-with-github/#github_from_pr) and
+[`--include-easyblocks-from-pr`](https://docs.easybuild.io/integration-with-github/#github_include_easyblocks_from_pr)
+to pull in changes required to make the installation work correctly in the EESSI build environment,
+but only if that is strictly required.
+
+!!! note
+
+    This restriction may be relaxed later to also allow adding software that is not supported yet in the latest
+    EasyBuild release, or to allow for installing software with other tools.
+
+## Supported compiler toolchains
+
+A [compiler toolchain](https://docs.easybuild.io/terminology/#toolchains) that is still supported by the latest
+EasyBuild must be used for building the software.
+
+More information on deprecated toolchains in EasyBuild is available
+[here](https://docs.easybuild.io/deprecated-easyconfigs/#deprecated_easyconfigs_toolchains).
```

Comment:
I'd say that any toolchain still supported in EasyBuild is eligible to be included in EESSI, at least in the initial version of the policy.

Going back to older toolchains is likely going to be significantly more painful, if only to get even the installation of GCC to work, so this is sort of self-regulating...

Also, we can't really use a statement like "Only toolchains already available in the EESSI repository may be used.", because then adding toolchains (however recent) would be against the contribution policy? :)

Reply:
I think it should be up to the EESSI maintainers what toolchains are and are not supported by a particular EESSI release, this can be handled via our hook, see for example https://github.com/easybuilders/JSC/blob/2024/Custom_Hooks/eb_hooks.py#L570-L640

Reply:
Allowing someone to use a new toolchain is a big step, as obviously this is extremely likely to trigger a massive number of new software packages (and maintainer effort).